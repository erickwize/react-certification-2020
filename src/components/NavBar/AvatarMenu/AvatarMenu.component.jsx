import React from 'react';
import './AvatarMenu.styles.css';
import {useHistory} from 'react-router';
import {useAuth} from '../../../providers/Auth';
import {mockedUser} from '../../../pages/Login/api';
import avatarImagen from './avatar.jpg';

export default function AvatarMenu(){
    const { authenticated, logout } = useAuth();
    const history = useHistory();

    const gotoLogin = () => {
        if(!authenticated){
            history.push('/login');
        }else{
            logout();
        }
    }
    return(
        <div onClick={gotoLogin}>
            {
                <img src={authenticated?mockedUser.avatarUrl:avatarImagen} alt={mockedUser.id} className="avatarMenu"/> 
            }
        </div>
        
    );
}