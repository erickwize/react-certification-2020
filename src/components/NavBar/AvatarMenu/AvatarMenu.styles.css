.avatarMenu{
    font-size: 43px;
    color: rgba(255, 255, 255, 0.4);
    margin-left: 10px;
    margin-right: 10px;
    border-radius: 50%;
    width: 40px;
}

@media only screen and (max-width: 650px) {
    .avatarMenu{
        position: absolute;
        right: 8px;
        top: 4px;
        width: 35px;
    }
}

@media only screen and (max-width: 800px) {
    .avatarMenu{
        font-size: 35px;
    }    
}