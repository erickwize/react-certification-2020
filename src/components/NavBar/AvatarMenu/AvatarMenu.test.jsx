import React from 'react';
import {render, screen} from '@testing-library/react';
import AvatarMenu from './AvatarMenu.component';
import AuthProvider from '../../../providers/Auth';

describe('Avatar Menu', () => {
    it('renders', () => {
        render(
        <AuthProvider>
            <AvatarMenu/>
        </AuthProvider>
        )
        const image = screen.getByAltText("123");
        expect(image).toBeInTheDocument();
    });
    
    test('starts as not logged in', () => {
        render(
        <AuthProvider>
            <AvatarMenu/>
        </AuthProvider>
        )
        const image = screen.getByAltText("123");
        expect(image.src.endsWith("avatar.jpg")). toBeTruthy();
    });

    test('matches snapshot', () => {
        render(
        <AuthProvider>
            <AvatarMenu/>
        </AuthProvider>
        )
        const image = screen.getByAltText("123");
        expect(image).toMatchSnapshot();
    })
})