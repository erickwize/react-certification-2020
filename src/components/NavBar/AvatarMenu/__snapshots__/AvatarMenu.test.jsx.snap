// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar Menu matches snapshot 1`] = `
<img
  alt="123"
  class="avatarMenu"
  src="avatar.jpg"
/>
`;
